src/index/hgraph_index.cpp

```diff
@@ -134,26 +134,26 @@ HGraphIndex::knn_search(const DatasetPtr& query,
         // check query vector
         CHECK_ARGUMENT(query->GetNumElements() == 1, "query dataset should contain 1 vector only");
 
-        auto ep = this->entry_point_id_;
+        InnerSearchParam search_param;
+        search_param.ep_ = this->entry_point_id_;
+        search_param.ef_ = 1;
```

Comment:
It is best to set the default value of ef to k.

Reply:
1 for high level graph